io/net/bonding.py

```diff
@@ -470,8 +459,17 @@ def test_cleanup(self):
                     self.log.warn("unable to bring to original state in peer")
                 time.sleep(self.sleep_time)
         self.error_check()
+
+        try:
+            NetworkInterface(self.bond_name, self.localhost).restore_from_backup()
```

Comment:
.restore_from_backup() should be called only when we have .saved() and .bring_up() has been called.. @smruti77 Please clarify here.

I mean if there is no sysfs bond file saved, than  why restore backup.

Reply:
also we are not creating sysfs bond configuration file in the bond_setup test ...

Reply:
We have a ifcfg-bond0 file created for all our bonding tests. It is defined in 
 def setup_ip(self)

We are creating bond master and slave using sysfs but the test flow ends up having a ifcfg-bond0 file created everytime. Also the format of this ifcfg-bond0 is like any other ifcfg-eth* file.